server/src/main/java/io/deephaven/server/table/TableServiceGrpcImpl.java

```diff
@@ -347,21 +347,19 @@ public void batch(final BatchTableRequest request,
                         errorInfo += " cause: " + cause.getMessage();
                         firstFailure.compareAndSet(null, (StatusRuntimeException) cause);
                     }
-                    final String finalErrorInfo = errorInfo;
-                    safelyExecuteLocked(responseObserver,
-                            () -> responseObserver.onNext(ExportedTableCreationResponse.newBuilder()
-                                    .setResultId(resultId)
-                                    .setSuccess(false)
-                                    .setErrorInfo(finalErrorInfo)
-                                    .build()));
+                    final ExportedTableCreationResponse response = ExportedTableCreationResponse.newBuilder()
+                            .setResultId(resultId)
+                            .setSuccess(false)
+                            .setErrorInfo(errorInfo)
+                            .build();
+                    safelyExecuteLocked(responseObserver, () -> responseObserver.onNext(response));
```

Comment:
We've got so many of these... I almost want to just have a wrapper/subtype of StreamObserver that does these for us, like onCompleted(Response) to do both onNext and onCompleted safely+locked, and auto-wrap both onNext and onCompleted with the lock and safe wrapper.